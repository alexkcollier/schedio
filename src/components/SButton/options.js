export {
  colors,
  allShadeOptions,
  colorNames,
} from '@/utils/component-options'

export const sizes = [
  'small',
  'regular',
  'large',
]

export const types = [
  'outlined',
  'text',
]
