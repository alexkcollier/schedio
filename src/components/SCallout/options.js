export const calloutTypes = ['', 'success', 'warning', 'error']
