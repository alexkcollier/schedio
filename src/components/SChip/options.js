import { props } from '@/assets/styles/tokens/dist/tokens.raw.json'

export const colors = [
  ...Object.values(props).reduce((colorList, { category, name }) => {
    if (category === 'background-color') colorList.add(name.split('-')[0])

    return colorList
  }, new Set()),
].sort()
