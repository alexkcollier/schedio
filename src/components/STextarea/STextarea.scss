.textarea {
  @include control;
  @include control-with-icons;
}
