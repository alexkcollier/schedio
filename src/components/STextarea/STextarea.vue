<template>
  <SLabel
    :for="id"
    :label="label"
    :is-inline="isInline"
    :is-fullwidth="isFullwidth"
    :is-required="required"
    :hide-optional="hideOptional"
  >
    <div class="textarea__container">
      <textarea
        :id="id"
        v-bind="$attrs"
        :class="{
          'textarea--invalid': isInvalid,
          'textarea--has-icon-left': !!iconLeft,
          'textarea--has-icon-right': !!iconRight,
        }"
        :name="_name"
        :placeholder="placeholder"
        :required="required"
        :rows="$attrs.rows || 10"
        class="textarea"
        v-on="listeners"
      />

      <SIcon
        v-if="iconLeft"
        :icon="iconLeft"
        :color="iconLeftColor"
        class="textarea__icon textarea__icon--left"
      />

      <SIcon
        v-if="iconRight"
        :icon="iconRight"
        :color="iconRightColor"
        class="textarea__icon textarea__icon--right"
      />
    </div>
  </SLabel>
</template>

<script>
import ControlIcons from '@/mixins/ControlIcons.mixin'
import InputText from '@/mixins/InputText.mixin'

export default {
  name: 'STextarea',

  mixins: [InputText, ControlIcons],
}
</script>

<style lang="scss">
.textarea {
  @include control;
  @include control-with-icons;
}
</style>
