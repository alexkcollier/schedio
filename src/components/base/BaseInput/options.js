export const allowed = ['email', 'text', 'password', 'url', 'search', ''].sort()
