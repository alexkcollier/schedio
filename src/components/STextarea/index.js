import STextarea from './STextarea.vue';

export { STextarea };

export default { install: Vue => Vue.component(STextarea.name, STextarea) };
