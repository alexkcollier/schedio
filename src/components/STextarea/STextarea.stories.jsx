import { SHeading } from '@/components/SHeading'
import { colors as iconColors } from '@/components/SIcon/options'
import { STextarea } from '@/components/STextarea'
import PropList from '@/docs/PropList'
import { withAttrsAsProps, withUnboundAttrs } from '@/mixins/stories/form-fields'
import { boolean, number, select, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/vue'
import icons from 'feather-icons/dist/icons.json'

const iconList = Object.keys(icons)
const iconProp = side => select(`icon-${side}`, ['', ...iconList], '', 'Optional Props')
const iconColor = side =>
  select(`icon-${side}-color`, ['', ...iconColors], 'green', 'Optional Props')

storiesOf('Components/Inputs.STextarea', module)
  .addParameters({ jest: 'STextarea' })
  .add('Textarea', () => {
    return {
      props: {
        props: {
          default: () => ({
            ...withAttrsAsProps({
              id: 'base-textarea',
              name: 'base-textarea',
              label: 'Base Textarea'
            }),
            placeholder: text('placeholder', 'Placeholder text', 'Required Props'),
            isInline: boolean('is-inline', false, 'Optional Props'),
            iconLeft: iconProp('left'),
            iconLeftColor: iconColor('left'),
            iconRight: iconProp('right'),
            iconRightColor: iconColor('right')
          })
        },

        attrs: {
          default: () => ({
            ...withUnboundAttrs(),
            numberOfRows: number('rows (default: 10)', 10, {}, '$attrs')
          })
        }
      },
      render(h) {
        const { props, attrs } = this.$props

        return (
          <div>
            <SHeading level="1">Textarea</SHeading>
            <p>Textareas are used to collect user provided text data is relatively long.</p>

            <SHeading level="2">Example</SHeading>
            <STextarea {...{ props }} {...{ attrs }} />

            <PropList component={STextarea} />
          </div>
        )
      }
    }
  })
