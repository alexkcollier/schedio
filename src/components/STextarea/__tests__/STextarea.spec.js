import { colors as iconColors } from '@/components/SIcon/options'
import STextarea from '@/components/STextarea/STextarea.vue'
import { shallowMount } from '@vue/test-utils'
import icons from 'feather-icons/dist/icons.json'

const iconList = Object.keys(icons)

describe('STextarea.vue', () => {
  const errorSpy = jest.spyOn(global.console, 'error').mockImplementation(() => {})
  const inputEvent = jest.fn()
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(STextarea, {
      propsData: {
        id: 'textarea',
        name: 'textarea',
        label: 'textarea'
      },
      listeners: {
        input: inputEvent
      }
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('requires required props', () => {
    // mounting component without required props
    shallowMount(STextarea)
    expect(errorSpy).toBeCalled()
    expect(errorSpy.mock.calls[0][0]).toContain('[Vue warn]: Missing required prop')
  })

  it('can be required', () => {
    wrapper.setProps({ required: true })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can be invalid', () => {
    wrapper.setProps({ isInvalid: true })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can have a placeholder', () => {
    wrapper.setProps({ placeholder: 'some placeholder' })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('accepts user input', () => {
    const textarea = wrapper.find('textarea')
    textarea.setValue('text')
    expect(wrapper.emitted('input')[0][0]).toBe('text')
    expect(textarea.element.value).toBe('text')
  })

  it('can have an icon on the left', () => {
    wrapper.setProps({ iconLeft: iconList[0] })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can have an icon on the right', () => {
    wrapper.setProps({ iconRight: iconList[0] })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can set each icon color', () => {
    wrapper.setProps({
      iconLeftColor: iconColors[0],
      iconRightColor: iconColors[1]
    })

    expect(errorSpy).not.toBeCalled()
    expect(wrapper.props('iconLeftColor')).not.toBe(wrapper.props('iconRightColor'))
    expect(wrapper.html()).toMatchSnapshot()
  })
})
