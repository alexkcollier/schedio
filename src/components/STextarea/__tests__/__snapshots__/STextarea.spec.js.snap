// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STextarea.vue can be invalid 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" required="required" rows="10" class="textarea textarea--invalid"></textarea>
    <!---->
    <!---->
  </div>
</label>
`;

exports[`STextarea.vue can be required 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" required="required" rows="10" class="textarea"></textarea>
    <!---->
    <!---->
  </div>
</label>
`;

exports[`STextarea.vue can have a placeholder 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="some placeholder" required="required" rows="10" class="textarea"></textarea>
    <!---->
    <!---->
  </div>
</label>
`;

exports[`STextarea.vue can have an icon on the left 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" required="required" rows="10" class="textarea textarea--has-icon-left"></textarea> <svg viewBox="0 0 24 24" class="icon textarea__icon textarea__icon--left icon--color-green">
      <polyline points="22 12 18 12 15 21 9 3 6 12 2 12"></polyline>
    </svg>
    <!---->
  </div>
</label>
`;

exports[`STextarea.vue can have an icon on the right 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" required="required" rows="10" class="textarea textarea--has-icon-right"></textarea>
    <!----> <svg viewBox="0 0 24 24" class="icon textarea__icon textarea__icon--right icon--color-green">
      <polyline points="22 12 18 12 15 21 9 3 6 12 2 12"></polyline>
    </svg></div>
</label>
`;

exports[`STextarea.vue can hide the optional tag 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" rows="10" class="textarea"></textarea>
    <!---->
    <!---->
  </div>
</label>
`;

exports[`STextarea.vue can set each icon color 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" required="required" rows="10" class="textarea"></textarea>
    <!---->
    <!---->
  </div>
</label>
`;

exports[`STextarea.vue renders correctly 1`] = `
<label class="label" for="textarea">
  <div>
    textarea
    <!---->
  </div>
  <div class="textarea__container"><textarea id="textarea" placeholder="" required="required" rows="10" class="textarea"></textarea>
    <!---->
    <!---->
  </div>
</label>
`;
